lib/select.ts

```diff
@@ -50,43 +53,54 @@ export function select<T>(
   // Computed contains the IOptionFull of the obs value.
   const selected = Computed.create(null, obs, (use, val) => {
     const option = callback().find(_op => val === getOptionFull(_op).value);
-    return option ? getOptionFull(option) : ({value: null, label: ''} as IOptionFull<null>);
+    return option ? getOptionFull(option) : ({value: null, label: options.defaultLabel || ""} as IOptionFull<null>);
   });
 
-  const container: Element = cssSelectBtn({tabIndex: '0', class: options.buttonCssClass || ''},
+  // Select button and associated event/disposal DOM.
+  const selectBtn: Element = cssSelectBtn({tabIndex: '0', class: options.buttonCssClass || ''},
     dom.autoDispose(selected),
-    dom.domComputed(selected, sel =>
-      renderOption(sel.label ? sel : {value: null, label: options.defaultLabel || ""})),
+    dom.domComputed(selected, sel => renderOption(sel)),
     options.buttonArrow,
     dom.on('keydown', (ev) => {
       const sel = keyState.add(ev.key);
       if (sel) { obs.set(sel.value); }
-    }),
-    (elem) => baseElem((...args) => Select.create(...args), elem, () => [
-      (elem) => stretchMenuToContainer(elem, container),
-      dom.forEach(callback(), (option) => {
-        const obj: IOptionFull<T> = getOptionFull(option);
-        // Note we only set 'selected' when an <option> is created; we are not subscribing to obs.
-        // This is to reduce the amount of subscriptions, esp. when number of options is large.
-        return menuItem(() => { obs.set(obj.value); },
-          Object.assign({ disabled: obj.disabled, selected: obj.value === obs.get() },
-            obj.disabled ? { class: 'disabled'} : {}),
-          renderOption(obj)
-        );
-      }),
-    ], {
-      menuCssClass: options.menuCssClass,
-      trigger: [(triggerElem: Element, ctl: PopupControl) => {
-        dom.onElem(triggerElem, 'click', () => ctl.toggle()),
-        dom.onKeyElem(triggerElem as HTMLElement, 'keydown', {
-          ArrowDown: () => ctl.open(),
-          ArrowUp: () => ctl.open()
-        })
-      }],
-      selectLabelOnOpen: () => selected.get().label
     })
   );
-  return container;
+
+  // Options to pass into the Select class.
+  const selectOptions: ISelectOptions = {
+    ...defaultMenuOptions,
+    menuCssClass: options.menuCssClass,
+    trigger: [(triggerElem: Element, ctl: PopupControl) => {
+      dom.onElem(triggerElem, 'click', () => ctl.toggle()),
+      dom.onKeyElem(triggerElem as HTMLElement, 'keydown', {
+        ArrowDown: () => ctl.open(),
+        ArrowUp: () => ctl.open()
+      })
+    }],
+    selectLabelOnOpen: () => selected.get().label
+  };
+
+  // DOM content of the open select menu.
+  const selectContent = [
```

Comment:
Shouldn't this be a function? (As it is, callback() is only called once, I think, and `Select.create` is using the same result each time.)

Reply:
Yes